stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java

```diff
@@ -61,8 +62,16 @@ public static Object nativeRespond(ObjectValue connectionObj, ObjectValue outbou
         HttpCarbonMessage inboundRequestMsg = HttpUtil.getCarbonMsg(connectionObj, null);
         Strand strand = Scheduler.getStrand();
         DataContext dataContext = new DataContext(strand, new NonBlockingCallback(strand), inboundRequestMsg);
-        HttpCarbonMessage outboundResponseMsg = HttpUtil
-                .getCarbonMsg(outboundResponseObj, HttpUtil.createHttpCarbonMessage(false));
+        if (isDirtyResponse(outboundResponseObj)) {
+            String errorMessage = "Couldn't complete the respond operation as the response has been already used.";
+            HttpUtil.sendOutboundResponse(inboundRequestMsg, HttpUtil.createErrorMessage(errorMessage, 500));
+            unBlockStrand(strand);
+            log.debug(errorMessage);
```

Comment:
Should we provide more details with this error log to trace this request ? (May be `inboundRequestMsg.getSequenceId()`)

Reply:
We couldn't expose the inboundRequestMsg.getSequenceId() to the ballerina and it will not be useful to the user as they can't understand this.

Reply:
This log is for troubleshooting purposes right? Same is used to track other debug log messages [1]. @chamil321 Any thoughts? Btw, don't we need to check `log.isDebugEnabled()` before having debug logs here?

[1] https://github.com/ballerina-platform/ballerina-lang/blob/e2165dc989175d19a3abeb4b0a8dbbd77f5ffb8a/stdlib/http/src/main/java/org/ballerinalang/net/http/nativeimpl/connection/Respond.java#L96

Reply:
Fixed

Reply:
+1 Let's add the change to the master branch as well

Reply:
Fixed in #21090